src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -466,11 +466,11 @@ protected String generateAssertionsForPublicFields(Set<FieldDescription> fields,
   private String assertionContentForField(FieldDescription field, ClassDescription classDescription) {
     final String fieldName = field.getName();
     final String fieldNameCap = capitalize(field.getName());
-    if (classDescription.getGettersDescriptions().contains(new GetterDescription(fieldName, "get" + fieldNameCap,
-                                                                                 field.getTypeDescription(),
-                                                                                 Collections.<TypeName> emptyList()))) {
+    
+    if (classDescription.findGetterDescriptionForField(field) != null) {
```

Comment:
I agree with the name. For return type, I think it could be useful later to have it returning the value. If we were on Java 8 I'd rather return an `Optional<>`, though. Thought?s